database/database.go

```diff
@@ -1,36 +1,46 @@
-package nosql
+package database
 
 import (
-	"errors"
 	"fmt"
+
+	"github.com/pkg/errors"
```

Comment:
The global errors created using `github.com/pkg/errors` will only have the line where the error was created in the stack trace. We can use simple errors from the `"errors"` package and wrap them on the implementations.

The code that uses this package can check if is, for example, an ErrNotFound using the `Causer` interface or with the `IsErrNotFound` method.